--- a/packages/external/NOMAD/src/Eval_Point.cpp
+++ b/packages/external/NOMAD/src/Eval_Point.cpp
@@ -500,7 +500,7 @@
     int m = _bb_outputs.size();
     for ( int i = 0 ; i < m ; ++i ) {
         if ( _bb_outputs[i].is_defined() ) {
-#ifdef WINDOWS
+#if defined(WINDOWS) && !defined(__MINGW32__)
             if ( isnan ( _bb_outputs[i].value() ) )
                 return true;
 #else
--- a/packages/external/NOMAD/src/Eval_Point.hpp
+++ b/packages/external/NOMAD/src/Eval_Point.hpp
@@ -47,7 +47,7 @@
 #include "Cache_File_Point.hpp"
 #include "Set_Element.hpp"
 
-#ifdef WINDOWS
+#if defined(WINDOWS) && !defined(__MINGW32__)
 #ifndef isnan
 inline bool isnan ( double x ) { return x != x; }
 #endif
