diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 874f607b1..a1a42bc7e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -326,6 +326,9 @@ if(DAKOTA_PYTHON_DIRECT_INTERFACE)
     add_definitions("-DDAKOTA_PYTHON_NUMPY")
   endif()
 
+  # Add Python include directories and libraries to the build
+  list(APPEND EXT_TPL_INCDIRS ${Python3_INCLUDE_DIRS})
+
 endif()
 
 
@@ -1008,7 +1011,7 @@ if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_
 endif()
 # When using Boost imported targets, we only link libraries using them,
 # then rely on transitive library linking from CMake
-target_link_libraries(dakota_src dakota_src_fortran ${DAKOTA_BOOST_TARGETS})
+target_link_libraries(dakota_src dakota_src_fortran ${Boost_LIBRARIES})
 # Dakota should always depend on util (consider removing option in DakotaOptions.cmamke
 target_link_libraries(dakota_src dakota_util)
 # Keep parser/IR as implementation-only link dependencies of libdakota_src.
@@ -1133,7 +1136,9 @@ endif()
 
 
 target_link_libraries(dakota ${DAKOTA_ALL_LIBS})
-target_link_libraries(dakota dakota_parser_lib)
+target_link_libraries(dakota dakota_parser_lib dakota_ir)
+find_package(Threads REQUIRED)
+target_link_libraries(dakota Threads::Threads ${CMAKE_DL_LIBS} util ${Python3_LIBRARIES})
 DakotaApplyMPISettings(dakota)
 target_link_libraries(dakota_restart_util ${DAKOTA_ALL_LIBS})
 target_link_libraries(dakota_restart_util dakota_parser_lib)
@@ -1211,7 +1216,7 @@ set(Dakota_LIBRARIES ${DAKOTA_LIBS} ${DAKOTA_PKG_LIBS} ${DAKOTA_EXTPROJ_LIBS}
 
 # Libraries that will be on the system
 # We add Boost libraries by path/name here for Makefile.export purposes
-set(Dakota_TPL_LIBRARIES ${EXT_TPL_LIBS} ${Boost_LIBRARIES}
+set(Dakota_TPL_LIBRARIES ${EXT_TPL_LIBS}
   CACHE INTERNAL "Dakota TPL libraries") 
  
 message(STATUS "Dakota_LIBRARIES: ${Dakota_LIBRARIES}")
